pennylane/interfaces/batch/__init__.py

```diff
@@ -324,15 +324,16 @@ def cost_fn(params, x):
     if expand_fn == "device":
         expand_fn = lambda tape: device.expand_fn(tape, max_expansion=max_expansion)
 
-    if gradient_fn is None:
+    # don't unwrap if its an interface device
+    if (gradient_fn is None) and ("passthru_interface" not in device.capabilities()):
         with qml.tape.Unwrap(*tapes):
             res = cache_execute(batch_execute, cache, return_tuple=False, expand_fn=expand_fn)(
                 tapes
             )
 
         return batch_fn(res)
 
-    if gradient_fn == "backprop" or interface is None:
+    if gradient_fn == "backprop" or interface is None or gradient_fn is None:

```

Comment:
For readability: adding `gradient_fn is None` to the condition can be challenging to understand at first if the reader only looks at this specific line. Due to the previous `if` statement on line 328, in practice `(gradient_fn is None)` === `(gradient_fn is None) and ("passthru_interface" in device.capabilities())`.

Could this info be either commented or somehow conveyed by creating well-named objects that are then used in the condition?